src/asynk/async_worker.rs

```diff
@@ -88,14 +94,38 @@ where
 
     pub async fn run_tasks(&mut self) -> Result<(), Error> {
         loop {
+            //fetch task
             match self
                 .queue
                 .fetch_and_touch_task(Some(self.task_type.clone()))
                 .await
             {
                 Ok(Some(task)) => {
-                    self.sleep_params.maybe_reset_sleep_period();
-                    self.run(task).await?
+                    let actual_task: Box<dyn AsyncRunnable> =
+                        serde_json::from_value(task.metadata.clone()).unwrap();
+
+                    // check if task is scheduled or not
+                    match actual_task.cron() {
+                        None => {
+                            self.sleep_params.maybe_reset_sleep_period();
```

Comment:
this sleep reset and run can be moved outside of this match

so in the match you will only schedule the next task

Reply:
[comments suggestions](https://github.com/ayrat555/fang/pull/71/commits/4eb86a2e2a6a2e33153736332de9191e4a162d56)